File: src/main/java/com/autosale/shop/model/ExceptionResponse.java
package com.autosale.shop.model;

import lombok.Value;
import org.springframework.http.HttpStatus;

import java.sql.Timestamp;
import java.time.LocalDateTime;

@Value
public class ExceptionResponse {
    Timestamp timestamp;
    int status;
    HttpStatus error;
    String message;

    public ExceptionResponse(HttpStatus error, String message) {
        this.timestamp = Timestamp.valueOf(LocalDateTime.now());
        this.status = error.value();
        this.error = error;
        this.message = message;
    }
}

File: src/main/java/com/autosale/shop/model/ExceptionResponse.kt
package com.autosale.shop.model

import org.springframework.http.HttpStatus
import java.sql.Timestamp
import java.time.LocalDateTime

data class ExceptionResponse(
    val timestamp: Timestamp,
    val status: Int,
    val error: HttpStatus,
    val message: String
) {
    constructor(error: HttpStatus, message: String) : this(
        Timestamp.valueOf(LocalDateTime.now()),
        error.value(),
        error,
        message
    )
}
